R/zyp.R

```diff
@@ -185,12 +185,17 @@ zyp.yuepilon <- function(y, x=1:length(y), conf.intervals=TRUE, preserve.range.f
   yt <- yt.prime[1:(n-1)] + trend * t.prime
   dmap.prime <- which(!is.na(yt))
   ytnm <- as.numeric(yt[dmap.prime])
+  t.prime <- t.prime[dmap.prime]
 
   # Calculate the Mann-Kendall test for significance on the blended series
-  Kend <- Kendall(t.prime[dmap.prime],ytnm)
+  Kend <- Kendall(t.prime,ytnm)
   tau <- Kend[1]
   Bsig <- Kend[2]
 
+  # Calculate slope on pre-whitened data

```

Comment:
Yes, as the initial trend is used in the pre-whitening routine in lines 172-188.